packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2163,20 +2163,37 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
 
     final TextStyle hintStyle = _getInlineHintStyle(themeData, defaults);
     final String? hintText = decoration.hintText;
-    final Widget? hint = hintText == null ? null : AnimatedOpacity(
-      opacity: (isEmpty && !_hasInlineLabel) ? 1.0 : 0.0,
-      duration: decoration.hintFadeDuration ?? _kHintFadeTransitionDuration,
-      curve: _kTransitionCurve,
-      child: Text(
+    final bool hintIgnoreHeightOnInput = decoration.hintIgnoreHeightOnInput;
+    Widget? hint;
+    if (hintText != null) {
+      final bool showHint = isEmpty && !_hasInlineLabel;
+      final Text hintTextWidget = Text(
         hintText,
         style: hintStyle,
         textDirection: decoration.hintTextDirection,
         overflow: hintStyle.overflow ?? (decoration.hintMaxLines == null ? null : TextOverflow.ellipsis),
         textAlign: textAlign,
         maxLines: decoration.hintMaxLines,
-      ),
-    );
-
+      );
+      hint = hintIgnoreHeightOnInput ? AnimatedSwitcher(
+        duration: decoration.hintFadeDuration ?? _kHintFadeTransitionDuration,
+        transitionBuilder: (Widget child, Animation<double> animation) {
+          return FadeTransition(
+            opacity: CurvedAnimation(
+              parent: animation,
+              curve: _kTransitionCurve,
+            ),
+            child: child,
+          );
+        },
+        child: showHint ? hintTextWidget : const SizedBox.shrink(),
```

Comment:
Does it work if we only change the leaf here based on the flag? i.e.:
```dart
child: !hintIgnoreHeightOnInput || showHint ? hintTextWidget : const SizedBox.shrink(),
 ```
So we can get rid of the `AnimatedOpacity` branch?

Reply:
@LongCatIsLooong I’m sorry, but I tried it, but it seems that it can’t hide the hint, and there’s no animation effect.